app/src/main/java/com/ubergeek42/WeechatAndroid/relay/Lines.kt

```diff
@@ -103,6 +135,9 @@ class Lines {
             if (status == Status.Init && unfiltered.size > 0) addLast(SquiggleLine())   // invisible
         }
 
+        addDateLine(unfiltered, line.timestamp)
```

Comment:
this creates two different date change lines. not only this does the job twice but the UI will treat these two lines as different which can result in weird animations

Reply:
well the filtered version could squash consecutie date changes while the unfiltered version does not. Can you be more explicit about hte weird animations?

Reply:
the UI treats lines as different if they have different pointers. this creates two lines with different pointers. so when you toggle line filtering, this should result in RecyclerView's "item changed" animation i think

re: squashing -- why filtered only?

Reply:
> the UI treats lines as different if they have different pointers. this creates two lines with different pointers. so when you toggle line filtering, this should result in RecyclerView's "item changed" animation i think

so would it work for you if I give them the same id?

> re: squashing -- why filtered only?

that was only an example why the lines could be different. If there where no unfiltered lines in between both would be the same.

Reply:
after re-reading this i understand that in fact there can be several unfiltered date change lines for one filtered. so it's not always possible to say that this is the same object. so having different ids would make sense in some circumstances. in others, however, when the dates are equal, it would make sense to use the same id. i guess